src/components/atoms/button.tsx

```diff
@@ -1,7 +1,30 @@
-import React from 'react'
+import React, { Children, ReactNode } from "react";
+import { ButtonStyles } from "../../common/styles";
+
+type ButtonProps = {
+  text: string;
+  className?: string;
+  children?: ReactNode;
+  variant?: "container" | "outline" | "text";
+  func?: "edit" | "delete" | "add";
+  action?: () => {};
+};
+
+export const Button = ({
+  text,
+  className,
+  children,
+  variant = "container",
+  func = "edit",
+}: ButtonProps) => {
+  const btnStyles = new ButtonStyles(variant);
+  btnStyles.setFunctionType(func);
+  const styles = btnStyles.getButtonStyle();
```

Comment:
We usually keeps this type of styling for a component inside the component so. 
Tts close by. I also can't see the code for this ButtonStyles